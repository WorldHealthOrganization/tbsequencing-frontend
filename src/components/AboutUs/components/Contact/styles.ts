import { css } from '@emotion/react';
import { fontSizes } from '../../../typography/fontSizes';
import appColors from '../../../../styles/colors';

export const container = css({
  width: 200,
});

export const primaryText = css({
  fontSize: fontSizes.h3,
  color: appColors.neutral.white,
  lineHeight: '24px',
});

export const header = css({
  color: appColors.neutral.white,
  marginBottom: 8,
});

export const link = css({
  textDecoration: 'none',
  ':hover': {
    backgroundColor: appColors.secondary.whoBlue,
  },
  display: 'flex',
});
export const primaryTextRight = css({
  marginLeft: 3,
});
