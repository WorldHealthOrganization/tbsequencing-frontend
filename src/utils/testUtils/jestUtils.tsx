import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
