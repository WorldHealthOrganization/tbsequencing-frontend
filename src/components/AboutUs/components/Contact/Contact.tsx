/** @jsxImportSource @emotion/react */
import React from 'react';
import { SerializedStyles } from '@emotion/react';
import * as styles from './styles';
import H2 from '../../../typography/H2';
import PrimaryText from '../../../typography/PrimaryText';

interface Props {
  title: string;
  phone?: string;
  email: string;
  webSite: string;
  style?: SerializedStyles;
  linkLabel: string;
}

export const Contact = ({
  title,
  phone,
  email,
  webSite,
  style,
  linkLabel,
}: Props) => (
  <div css={[styles.container, style]}>
    <H2 style={styles.header}>{title}</H2>
    {
     phone && (
     <a css={styles.link} href={`tel:${phone}`}>
       <PrimaryText style={styles.primaryText}>Phone: </PrimaryText>
       <PrimaryText style={[styles.primaryText, styles.primaryTextRight]}>{phone}</PrimaryText>
     </a>
     )
    }

    <a css={styles.link} href={`mailto: ${email}`}>
      <PrimaryText style={[styles.primaryText, styles.primaryTextRight]}>{email}</PrimaryText>
    </a>

    <a css={styles.link} target="_blank" href={webSite} rel="noreferrer">
      <PrimaryText style={[styles.primaryText, styles.primaryTextRight]}>{linkLabel}</PrimaryText>
    </a>

  </div>
);

export default Contact;
