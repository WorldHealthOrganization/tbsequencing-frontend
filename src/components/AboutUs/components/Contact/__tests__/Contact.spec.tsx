import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../../../utils/testUtils/jestUtils';
import Contact from '../Contact';

describe('Contact component test', () => {
  test('renders Contact component without error', () => {
    render(
      <Contact />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <Contact />,
    );

    expect(tree).toMatchSnapshot();
  });
});
