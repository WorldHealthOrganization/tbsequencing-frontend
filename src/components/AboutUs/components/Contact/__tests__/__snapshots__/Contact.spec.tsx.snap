// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component test Should render correctly 1`] = `
<div
  className="css-vtpaov-Contact"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-17mv985-MuiTypography-root-BaseText"
  />
  <a
    className="css-1tbc68-Contact"
    href="mailto: undefined"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-zzhh7v-MuiTypography-root-BaseText"
    />
  </a>
  <a
    className="css-1tbc68-Contact"
    rel="noreferrer"
    target="_blank"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-zzhh7v-MuiTypography-root-BaseText"
    />
  </a>
</div>
`;
